qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -425,7 +425,11 @@ def _wait_for_submission(self, timeout=60):
                 raise JobTimeoutError(
                     "Timeout waiting for the job being submitted: {}".format(ex)
                 )
-
+            if (submit_info is None
```

Comment:
I think we could move the check just after checking `self._future` to keep the timeout signaling in the last place and remove the check of `submit_info` (I'm not sure if it's possible at all).

Reply:
Yep, I was going to suggest the same thing.

Reply:
ok sounds good

Reply:
Since `_future_captured_exception` is set and `None` is returned together I'm fine with just checking one, however it is `submit_info` begin `None` which causes a the different exception and "hides" the real one.